posts/Jan/2022-04-29-mapping-boolean-types-in-hibernate-orm-6.adoc

```diff
@@ -0,0 +1,145 @@
+= Mapping Boolean attributes in ORM 6
+:awestruct-tags: [ "Hibernate ORM" ]
+:awestruct-layout: blog-post
+:icons: font
+Jan Schatteman <jschatte@redhat.com>
+
+By default, Hibernate maps a Boolean value to a BOOLEAN column. However, not all databases have a built-in BOOLEAN datatype and not all applications will want to use BOOLEAN (legacy schema e.g.).
+
+Hibernate provides multiple ways to map Boolean value to other SQL types; this post explores the most common ways.
+
+.Implicit Boolean mapping
+====
+[source, JAVA, indent=0]
+----
+
+    @Basic
+    private boolean aBoolean;
+
+----
+====
+In the above example the Boolean attribute will be mapped to what is returned by the call to `JdbcTypeIndicators.getPreferredSqlTypeCodeForBoolean()`, which in turn reverts to either the value configured for the `AvailableSettings.PREFERRED_BOOLEAN_JDBC_TYPE` setting or, if that is not set, the value returned from the call to `Dialect.getPreferredSqlTypeCodeForBoolean()`, the default being `Boolean` in case a specific dialect doesn't override this.
```

Comment:
Stuff like `JdbcTypeIndicators.getPreferredSqlTypeCodeForBoolean()` is way to low-level for users.

The discussion here needs to be much more generalized IMO.  Something like -

```
How Hibernate chooses to map a boolean implicitly is determined by a series of cascading checks:

  1. It first checks whether the link:{javadoc-AvailableSettings}#PREFERRED_BOOLEAN_JDBC_TYPE[hibernate.type.preferred_boolean_jdbc_type] setting has been specified and, if so, uses it;
  2. Next it consults the link:{javadoc-Dialect}#getPreferredSqlTypeCodeForBoolean[Dialect] to see if it reports a preferred SQL type and, if so, uses it;
  3. Otherwise it will use link:{javadoc-Types}#BOOLEAN[Types.BOOLEAN]
```